File: HW1/C++/HW1.cpp
#include <iostream>
#include <string>
#include <cctype>

using namespace std;

int main(int argc, char** argv) {
    
    string tmp;
    int i=0,j=0,len=0,isPal=1;
    char tmpr,tmpl;
    
    cout<<"Hint: Key Enter Directly to Exit!\n"<<endl
        <<"Please enter potential palindrome:"<<endl;

    //get line including space form stdin
    getline(cin, tmp);
    len = tmp.size();

    // Can Key Enter without other characters to Exit!
    while(len!=0)
    {        
        i=0; j=len-1; isPal = 1;

        while(i<len || j>=0)
        {
            while(i<len && !isalpha(tmp[i])) i++;
            while(j>=0 && !isalpha(tmp[j])) j--;
            
            // avoid the pointer i & j move over the range of String
            if(i>len || j<0) break;
            
            tmpr=tmp[i]; tmpl=tmp[j];
                                
            if(toupper(tmpr)!=toupper(tmpl))
            {
                    isPal = 0;
                    break;
            }
            
            i++; j--;
        }
        
        if(isPal==1)    cout<<"Yes\n"<<endl;
        else    cout<<"No\n"<<endl;
        
        //get line for next round in while loop
        cout<<"Please enter potential palindrome:"<<endl;
        getline(cin, tmp);
        len = tmp.size();    
    }

    return 0;
}

File: HW1/JAVA/hw1.java
import java.io.*;
import java.util.Scanner;


public class hw1
{
    public static void main(String args[])
    {
        String tmp;
        int i=0,j=0,len=0,isPal=1;;

        //scanner a line for System standard input
        Scanner scanner = new Scanner(System.in);


        //get a string including space and tab
        tmp = scanner.nextLine();
        len = tmp.length();

        while(len!=0)
        {
            i=0; j=len-1;
            isPal = 1;

            while(i<len || j>=0)
            {
                //If characters pointed are not alphabet, the pointers of comparison will move 
                while(!Character.isLetter(tmp.charAt(i))) i++;
                while(!Character.isLetter(tmp.charAt(j))) j--;

                if(tmp.charAt(i)!=tmp.charAt(j))
                {
                    isPal = 0;
                    break;
                }

                // move for next comparison
                i++; j--;
            }

            if(isPal == 1) System.out.println("Valid");
            else System.out.println("Invalid");

            tmp = scanner.nextLine();//read new line for next loop step
            len = tmp.length();
        }
    }
}